planner/memo/group.go

```diff
@@ -87,15 +88,40 @@ type Group struct {
 	EngineType EngineType
 }
 
+// buildLogicalProperty builds a new LogicalProperty for the new GroupExpr.
+// it also builds the unique key info and max one row info.
+func buildLogicalProperty(e *GroupExpr, oldSchema *expression.Schema) *property.LogicalProperty {
+	prop := &property.LogicalProperty{}
+	if oldSchema == nil {
+		return prop
+	}
+	newSchema := expression.NewSchema(oldSchema.Columns...)
+	childSchema := make([]*expression.Schema, len(e.Children))
+	childMaxOneRow := make([]bool, len(e.Children))
+	for i := range e.Children {
+		childSchema[i] = e.Children[i].Prop.Schema
+		childMaxOneRow[i] = e.Children[i].Prop.MaxOneRow
+	}
+	if len(childSchema) == 1 {
+		// For UnaryPlan(such as Selection, Limit ...), we can set the child's unique key as its unique key.
+		// If the GroupExpr is a schemaProducer, schema.Keys will be reset below in `BuildKeyInfo()`.
+		newSchema.Keys = childSchema[0].Keys
```

Comment:
It will be cleared when call `BuildKeyInfo`?

Reply:
Only `logicalSchemaProducer` will clear it.